addons/common/fnc_isTurnedOut.sqf

```diff
@@ -6,75 +6,81 @@ Description:
-SCRIPT(isTurnedOut);
-
 params ["_unit"];
 
-private _turn = false;
-private _out = false;
+if (isNull objectParent _unit) exitWith {true};
+if (isTurnedOut _unit) exitWith {true};
+
 private _vehicle = vehicle _unit;
-private _cfg = configFile >> "CfgVehicles" >> (typeOf _vehicle);
-
-if (_vehicle != _unit) then {
-    if (driver _vehicle == _unit) then {
-        private _forceHideDriver = getNumber(_cfg >> "forceHideDriver");
-        _turn = (_forceHideDriver != 1);
-        _out = (_vehicle animationPhase "hatchDriver") > 0;
-    } else {
-        private _assignedRole = assignedVehicleRole _unit;
-        if (_assignedRole select 0 == "Turret") then {
-            private _turretPath = _assignedRole select 1;
-            private _turret = [_vehicle, _turretPath] call CBA_fnc_getTurret;
-
-            private _canHideGunner = getNumber(_turret >> "canHideGunner");
-            private _forceHideGunner = getNumber(_turret >> "forceHideGunner");
-            _turn = (_canHideGunner == 1 && _forceHideGunner == 0);
-
-            private _hatchAnimation = getText(_turret >> "animationSourceHatch");
-            _out = (_vehicle animationPhase _hatchAnimation) > 0;
-
-        } else {
-            if ((_assignedRole select 0) == "Cargo") then {
-                private _attenuateCargo = getArray(_cfg >> "soundAttenuationCargo");
-                if ((count _attenuateCargo) > 0) then {
-                    private _index = (count _attenuateCargo)-1; // wait for command to get cargo index
-
-                    if (_index > -1) then {
-                        if(_index > (count _attenuateCargo)-1) then {
-                            _index = (count _attenuateCargo)-1;
-                        };
-                        if((_attenuateCargo select _index) == 0) then {
-                            _out = true;
-                        };
-                    };
-                };
-            };
+private _role = toLower (_unit call CBA_fnc_vehicleRole);
+private _isFFV = fullCrew [_vehicle, "turret"] findIf {_x select 0 == _unit && {_x select 4}} != -1;
+
+// special condition for RHS car windows
+if (_vehicle isKindOf "rhsusf_m1025_w" && {_role in ["driver", "turret"]}) then {
+    //will override _isFFV if window is closed
+    _isFFV = switch (_vehicle getCargoIndex _unit) do {
+        case -1: { // not in cargo
+            _vehicle doorPhase "ani_window_1" > 0
+        };
+        case 0: { // cargo position 0
+            _vehicle doorPhase "ani_window_2" > 0
+        };
+        case 1: { // cargo position 1
+            _vehicle doorPhase "ani_window_4" > 0
+        };
+        default { // any other cargo position
+            _vehicle doorPhase "ani_window_3" > 0
         };
     };
-} else {
-    _out = true;
 };
-_out;
 
+// FFVs are always outside
+if (_isFFV) exitWith {true};
+
+if (_role isEqualTo "commander") exitWith {
+    // nothing atm but so _cfgAttCargo will not executed
+    false // return
+};
```

Comment:
This seems pointless. Should be deleted, because the last `false` will handle it.

Reply:
the soundAtt... cfg will not executed... but yeah it seams pointless ^^

Reply:
Instead, you do another check, which is in total more expensive, and makes everything harder to read.